### PR TITLE
Backport NgOptimizedImage to Angular 14

### PR DESCRIPTION
**Pull request by atcastle**
(New PR created for 14.3.x branch. Old PR: #49230)
This is a PR for the effort by the Chrome Aurora team to backport NgOptimizedImage from Angular 15 to Angular 14. As noted in the [updated docs](https://github.com/angular/angular/pull/49230/files#diff-0ef174a039c0c873463a218843de86cf2ece0d1829923b08db0464b251b0d678R1), this is a one-time special project to make these performance enhancements available to more Angular applications, and does not imply a change in Angular release procedure or that any additional features will be backported.

CC: @AndrewKushnir @pkozlowski-opensource @kara

### CODE REVIEW

File: aio/content/guide/image-directive.md [hunk 1]
```diff
@@ -1,48 +1,55 @@
-# Getting started with NgOptimizedImage
-
-<div class="alert is-important">
+Notice: Angular 14.3.0 includes a backported version of the NgOptimizeImage directive present in later versions of Angular. This was implemented by the Chrome Aurora team, as a special project to make performance enhancements available to more applications built on Angular. This does not reflect a shift in Angular release procedure and does not indicate that any additional features will be backported to earlier Angular versions.
```
Comment (kara) [inline aio/content/guide/image-directive.md:1]:
```suggestion
Notice: Angular 14.3.0 includes a backported version of the NgOptimizedImage directive present in later versions of Angular. This was implemented by the Chrome Aurora team, as a special project to make performance enhancements available to more applications built on Angular. This does not reflect a shift in Angular release procedure and does not indicate that any additional features will be backported to earlier Angular versions.
```

Comment (kara) [reply aio/content/guide/image-directive.md:1]: Do we want to wrap the notice in an "alert" div?

**Review by kara — DISMISSED** [review]
LGTM

File: aio/content/guide/image-directive.md (hunk 1 above)
Comment (atcastle) [reply aio/content/guide/image-directive.md:2]: Done

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 2]
```diff
@@ -365,11 +511,48 @@ export class NgOptimizedImage implements OnInit, OnChanges, OnDestroy {
...
+    const finalSrcs = filteredBreakpoints.map(
+        bp => `${this.callImageLoader({src: this.ngSrc, width: bp})} ${bp}w`);
+    return finalSrcs.join(', ');
+  }
+
+  private getFixedSrcset(): string {
+    const finalSrcs = DENSITY_SRCSET_MULTIPLIERS.map(multiplier => `${this.callImageLoader({
+                                                       src: this.ngSrc,
+                                                       width: this.width! * multiplier
+                                                     })} ${multiplier}x`);
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:546]: The auto-formatting is strange here, we can probably split this into 2 lines for better readability:
```suggestion
    const finalSrcs = DENSITY_SRCSET_MULTIPLIERS.map(multiplier => {
      const imgUrl = this.callImageLoader({src: this.ngSrc, width: this.width! * multiplier});
      return `${imgUrl} ${multiplier}x`;
    });
```

**Review by AndrewKushnir — DISMISSED** [review]
Looks great (only 1 tiny comment) 👍

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts (hunk 2 above)
Comment (atcastle) [reply packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts]: Good suggestion--done!

**Review by mgechev — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
**Caretaker note**: this PR **requires a special handling by the dev-infra team**, while it's ready for merge, I'm adding the "blocked" label for now to indicate that.

**Comment by devversion** [comment]
FYI: Merging manually via rebase strategy. This PR is only supposed into the target branch, so merge tool is not needed

**Pull request merged**